src/Core/Usecase/Post/Export.php

```diff
@@ -11,69 +11,155 @@
 
 namespace Ushahidi\Core\Usecase\Post;
 
+use Ushahidi\App\Repository\ExportJobRepository;
+use Ushahidi\App\Repository\Form\AttributeRepository;
+use Ushahidi\App\Repository\Post\ExportRepository;
+use Ushahidi\Core\Tool\FormatterTrait;
 use Ushahidi\Core\Usecase\SearchUsecase;
 
 class Export extends SearchUsecase
 {
+	private $payload;
+	private $session;
+	private $data;
+	private $postExportRepository;
+	private $exportJobRepository;
+	private $formAttributeRepository;
+
+
+	// provides setFormatter to assign the formatter the usecase will use
+	use FormatterTrait;
+
 	// - VerifyParentLoaded for checking that the parent exists
 	use VerifyParentLoaded;
 
+	public function setExportJobRepository(ExportJobRepository $repo)
+	{
+		$this->exportJobRepository = $repo;//service('repository.export_job');
+	}
+
+	public function setFormAttributeRepository(AttributeRepository $repo)
+	{
+		$this->formAttributeRepository = $repo; //service('repository.form_attribute');
+	}
+
+	public function setPostExportRepository(ExportRepository $repo)
+	{
+		$this->postExportRepository = $repo; //service('repository.posts_export');
+	}
+
 	/**
-	 * Get filter parameters that are used for paging.
-	 *
-	 * @return Array
+	 * @param $payload
+	 * @param null $job_filters
+	 * @return array
+	 * Construct a filters object
 	 */
-	protected function getPagingFields()
+	public function constructFilters($payload, $job_filters = null)
 	{
-		return [
-			'orderby' => 'created',
-			'order'   => 'asc',
-			'limit'   => null,
-			'offset'  => 0
+		// Set the baseline filter parameters
+		$filters = [
+			'limit' => $payload['limit'],
+			'offset' => $payload['offset'],
 		];
+		// Merge the export job filters with the base filters
+		if ($job_filters) {
+			$filters = array_merge($filters, $job_filters);
+		}
+		return $filters;
+	}
+
+	/**
+	 * @param $job
+	 * @param $filters
+	 * @return mixed
+	 * Construct a Search Data object to hold the search info
+	 */
+	public function constructSearchData($job, $filters)
+	{
+		$data = $this->data->get('search');
+
+		// Set the fields that should be included if set
+		if ($job->fields) {
+			$data->include_attributes = $job->fields;
+		}
+
+		// set the filters that should be used
+		foreach ($filters as $key => $filter) {
+			$data->$key = $filter;
+		}
+
+		return $data;
 	}
 
-	// Usecase
 	public function interact()
```

Comment:
I think this needs to look more at what the SearchUsecase already does... it seems like a chunk of this could reuse that, or at least follow a more standard pattern.